From: Philipp Zabel <p.zabel@pengutronix.de>
Date: Thu, 22 Jan 2015 11:03:35 +0100
Subject: [PATCH] [media] imx-ipuv3-csi: enable IPU1 CSI0 input link by default

For backwards compatibility reasons, let the IPU1 CSI0 link be enabled on
startup.

Signed-off-by: Philipp Zabel <p.zabel@pengutronix.de>
---
 drivers/media/platform/imx/imx-ipuv3-csi.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/imx/imx-ipuv3-csi.c b/drivers/media/platform/imx/imx-ipuv3-csi.c
index 3d62958f8977..496ca2db6757 100644
--- a/drivers/media/platform/imx/imx-ipuv3-csi.c
+++ b/drivers/media/platform/imx/imx-ipuv3-csi.c
@@ -1810,12 +1810,21 @@ static int ipucsi_vb2_init(struct ipucsi *ipucsi)
 static int ipucsi_async_init(struct ipucsi *ipucsi, struct device_node *node)
 {
 	struct device_node *rp;
+	unsigned int flag = 0;
 
 	rp = of_get_next_child(node, NULL);
 	if (!rp)
 		return 0;
 
-	ipucsi->link = ipu_media_entity_create_link(&ipucsi->subdev, 0, rp, 0);
+	/*
+	 * For backwards compatibility, let the IPU1 CSI0 link be enabled on
+	 * startup.
+	 */
+	if (ipu_get_id(ipucsi->ipu) == 0 && ipucsi->id == 0)
+		flag = MEDIA_LNK_FL_ENABLED;
+
+	ipucsi->link = ipu_media_entity_create_link(&ipucsi->subdev, 0, rp,
+						    flag);
 
 	if (IS_ERR(ipucsi->link)) {
 		ipucsi->link = NULL;
